src/components/form/RenameFileForm.vue

```diff
@@ -0,0 +1,126 @@
+<template>
+  <q-form
+    class="q-gutter-md rename-file-form"
+    data-cy="rename-file-form"
+    @submit="onSubmit"
+  >
+    <q-input
+      v-model="fileName"
+      filled
+      :label="$t(`page.modelizer.fileExplorer.rename.${file.isFolder ? 'folder' : 'file'}.input`)"
+      lazy-rules
+      :rules="[
+        (value) => notEmpty(t, value),
+        (value) => isValidFileLabel($t, value),
+        (value) => isUnique(
+          $t,
+          labels,
+          value,
+          'errors.fileExplorer.label.duplicate',
+        ),
+      ]"
+      data-cy="name-input"
+    />
+    <div class="flex row items-center justify-center">
+      <q-btn
+        icon="fa-solid fa-save"
+        :label="$t('actions.default.save')"
+        type="submit"
+        :loading="submitting"
+        color="positive"
+        data-cy="submit-button"
+      >
+        <template #loading>
+          <q-spinner-dots />
+        </template>
+      </q-btn>
+    </div>
+  </q-form>
+</template>
+
+<script setup>
+import { useI18n } from 'vue-i18n';
+import { onMounted, ref } from 'vue';
+import { notEmpty, isValidFileLabel, isUnique } from 'src/composables/QuasarFieldRule';
+import { gitRemove, readDir, rename } from 'src/composables/Project';
+import { Notify } from 'quasar';
+import FileEvent from 'src/composables/events/FileEvent';
+
+const props = defineProps({
+  projectName: {
+    type: String,
+    required: true,
+  },
+  file: {
+    type: Object,
+    required: true,
+  },
+});
+
+const emit = defineEmits(['file:rename']);
+const { t } = useI18n();
+const fileName = ref('');
```

Comment:
Wouldn't it be better to have the current (old) name of the file already displayed in the field, so that if we want to modify it we don't need to rewrite it entirely?
And by doing so, we know which file we are currently renaming

Reply:
done